netlify/netlify-plugin-nextjs
# fix: don't use ODB for routes that match middleware

### Summary

Routes that match middleware need to be served via SSR, because the middleware is run at the origin. This PR ensures that any routes that matches mdidleware is not served by an ODB, which was causing errors where results were persisted including the middleware.

### Test plan

1. Visit [the Deploy Preview](https://deploy-preview-1171--netlify-plugin-nextjs-demo.netlify.app/getStaticProps/withFallbackAndMiddleware/4/)
2. Check the network tab: there should be a header `x-middleware-date` inserted by middleware. Ensure that the date is up to date. Also ensure that `x-nf-render-mode` is `ssr`.

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

Fixes #1168, Fixes #1114

![capybara](https://pbs.twimg.com/media/FKHkiahUUBElzTN?format=jpg&name=small)

### Standard checks:




---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.
